README.md
# Skyworkz website

This website is built using:

 * Hugo 0.55+
 * Bootstrap 4
 * Font-awesome
 * Ion icons

## Usage

First, run `npm install` to install all required node modules. Then run `hugo` to build all assets. Finally, run `hugo serve` to access the website at http://localhost:1313.

## Contributing

This website makes heavily use of Hugo for the content management. There are some guidelines to follow:

 * Use SASS/SCSS. Don't start modifying the `.css` files, only the `.scss`.
 * Install modules using NPM. This keeps our source repository clean.
 * Use Bootstrap as much as possible. Bootstrap is pretty powerfull.

## Questions

If you have any questions, ask Bas :)


changelog.txt
--------------------------------------------------------------------------
Version v2.6.1 - 18 December, 2018
--------------------------------------------------------------------------
Unify v2.6.1 update is focused on vendors (libraries and frameworks) updates, improvements and some minor bug fixes.

- ENHANCED: New Layout Pages were added in Unify Admin Template (included Fixing Sidebar).
- ENHANCED: MegaMenu "close all" feature was added.
- UPDATED: Chosen
- UPDATED: Clipboard
- UPDATED: Get-size
- UPDATED: Imagesloaded
- UPDATED: jQuery
- UPDATED: Masonry
- UPDATED: Plyr
- UPDATED: Popper.js
- UPDATED: Tooltip.js
- UPDATED: jQuery.Peity
- FIXED: Topbar Header in iPad
- FIXED: Dropdown column in Top Navigation [IE]
- FIXED: Cubeportfolio "Load More"
- FIXED: MegaMenu
- FIXED: and other minor bug fixes.

--------------------------------------------------------------------------
Version v2.6.0 - 15 August, 2018
--------------------------------------------------------------------------
Unify v2.6.0 update is focused on new RTL package for One Pages, E-Commerce and Multi Pages demos (RTL package for Unify Main and Admin Template demos will be added later), new components for Admin Template package, vendors (libraries and frameworks) updates, improvements and some minor bug fixes.

- NEW: RTL version for E-Commerce, OnePages & Multipages
- NEW: Advanced Form Slider Controls (Admin Template)
- NEW: Summernote WYSIWYG Editor (Admin Template)
- NEW: Summernote WYSIWYG Editor (Admin Template)
- NEW: Datatables Examples (Admin Template)

- ENHANCED/FIXED: Overall minor fixes and improvements

- UPDATED: Bootstrap 4.1.3
- UPDATED: Cube Portfolio 4.4
- UPDATED: Revolution Slider 5.4.8

--------------------------------------------------------------------------
Version v2.5.1 - 24 February, 2018
--------------------------------------------------------------------------
Unify v2.5.1 update is focused on vendors (libraries and frameworks) updates, improvements and some minor bug fixes. The major update includes Bootstrap 4 production version switching from Bootstrap 4 Beta version.

- ENHANCED: Shortcode Accordions
- ENHANCED: Shortcode Layout Forms
- ENHANCED: Google Map on Lightbox
- ENHANCED: Charts in Admin Dashboard for instant change viewport
- ENHANCED: Complex Tables view in Admin Dashboard on the mobile resolution

- FIXED: Datepicker date range
- FIXED: Activity Sidebar for mobile devices in Admin Dashboard

- UPDATED: Bootstrap 4.0.0
- UPDATED: Cube Portfolio 4.3.1
- UPDATED: Revolution Slider 5.4.6.4

--------------------------------------------------------------------------
Version v2.5 - 07 February, 2018
--------------------------------------------------------------------------

Unify v2.5 does not include any major changes. This update is focused on a new package "Unify Admin Template" and some minor bug fixes.

- NEW: Introducing brand new Package - Admin Template. This package includes over 50 pages.
- NEW: Admin Template html/admin-template/..
 - dashboard-v1.html
 - dashboard-v2.html
 - dashboard-v3.html
 - dashboard-v4.html
 - header-static.html
 - sidebar-hide.html (on click to humberger menu)
 - layout-light.html
 - layout-dark-body-v2.html
 - layout-light-body-v2.html
 - app-profile.html
 - app-profile-biography.html
 - app-profile-interests.html
 - app-profile-mobile.html
 - app-profile-photos-and-videos.html
 - app-profile-payment-methods.html
 - app-profile-transactions.html
 - app-profile-security.html
 - app-profile-upgrade-plan.html
 - app-profile-invite.html
 - app-profile-connected-accounts.html
 - app-projects.html
 - app-chat.html
 - app-file-manager.html
 - app-contacts.html
 - forms-elemets-text-inputs.html
 - forms-elemets-textareas.html
 - forms-elemets-selects.html
 - forms-elemets-advanced-selects.html
 - forms-elemets-checkboxes-radios.html
 - forms-elemets-toggles.html
 - forms-elemets-file-inputs.html
 - forms-elemets-sliders.html
 - forms-elemets-tags.html
 - forms-elemets-ratings.html
 - forms-elemets-datepickers.html
 - forms-elemets-quantities.html
 - forms-validation-states.html
 - tables-basic.html
 - tables-complex.html
 - panel-variations.html
 - panel-options.html
 - notifications-colorful.html
 - notifications-light.html
 - notifications-dark.html
 - notifications-builder.html
 - metrics.html
 - ui-icons.html
 - pages-timeline.html
 - packages.html

- NEW: Chartist JS - examples are using in Admin Dashboard Pages
- NEW: Variety Widget Blocks (Comments, Calendar, Tickets, Activity etc.) - examples are used in Admin Dashboard Pages
- NEW: Select Forms html/admin-template/forms/forms-elemets-selects.html
- NEW: Advanced Select Forms html/admin-template/forms/forms-elemets-advanced-selects.html
- NEW: Text Inputs with Tags html/admin-template/forms/forms-elemets-tags.html
- NEW: Datepicker html/admin-template/forms/forms-elemets-datepickers.html
- NEW: Quantities Design html/admin-template/forms/forms-elemets-quantities.html
- NEW: Validations Designs html/admin-template/forms/forms-validation-states.html
- NEW: Table Designs html/admin-template/tables/tables-basic.html
- NEW: Complex Tables (Data Tables) html/admin-template/tables/tables-complex.html
- NEW: 12 Panel Variations html/admin-template/panels/panel-variations.html
- NEW: 8 Panel's Options html/admin-template/panels/panel-options.html
- NEW: Colorful Notification html/admin-template/notifications/notifications-colorful.html
- NEW: Light Notification html/admin-template/notifications/notifications-light.html
- NEW: Dark Notification html/admin-template/notifications/notifications-dark.html
- NEW: Notification's Builder html/admin-template/notifications/notifications-builder.html
- NEW: Timeline Design html/admin-template/pages/pages-timeline.html

- ENHANCED: Sticky Block in less content
- ENHANCED: Slick Carousel - added preloader
- ENHANCED: Missing animate.css into show/copy code

- FIXED: Header - Megamenu 5 columns in IE browsers

html/admin-template/assets/js/components/hs.side-nav - copy.js
/**
 * Side Nav wrapper.
 *
 * @author Htmlstream
 * @version 1.0
 * @requires
 *
 */

var isSideMenuMini = $.cookie('isSideMenuMini') ? true : '';

;
(function($) {
	'use strict';

	$.HSCore.components.HSSideNav = {
		/**
		 *
		 *
		 * @var Object _baseConfig
		 */
		_baseConfig: {
			afterOpen: function (){},
			afterClose: function (){}
		},

		/**
		 *
		 *
		 * @var jQuery pageCollection
		 */
		pageCollection: $(),

		/**
		 * Initialization of Side Nav wrapper.
		 *
		 * @param String selector (optional)
		 * @param Object config (optional)
		 *
		 * @return jQuery pageCollection - collection of initialized items.
		 */
		init: function(selector, config) {
			this.collection = selector && $(selector).length ? $(selector) : $();
			if (!$(selector).length) return;

			this.config = config && $.isPlainObject(config) ?
				$.extend({}, this._baseConfig, config) : this._baseConfig;

			this.config.itemSelector = selector;

			this.initSideNav();

			return this.pageCollection;
		},

		initSideNav: function() {
			//Variables
			var $self = this,
				config = $self.config,
				collection = $self.pageCollection;

			//Actions
			this.collection.each(function(i, el) {
				//Variables
				var $this = $(el),
					target = $this.data('hssm-target'),
					targetClass = $this.data('hssm-class'),
					bodyClass = $this.data('hssm-body-class'),
					isUnifyEffect = Boolean($this.data('hssm-fade-effect')),
					isShowAllExceptThis = Boolean($this.data('hssm-is-close-all-except-this')),
					$closedItems = [],
					windW = $(window).width();

				$(target).find('[data-hssm-target]').each(function(i) {
					if ($(this).parent().hasClass('u-side-nav--top-level-menu-item u-side-nav-opened')) {
						$closedItems.push($(this).data('hssm-target'));
					}
				});

				// if (isUnifyEffect) {
				// 	$this.on('click', function(e) {
				// 		e.preventDefault();
        //
				// 		if (!$(target).hasClass('toggled')) {
				// 			$self.unifyOpenEffect(target, targetClass, bodyClass);
				// 			$.cookie('isSideMenuMini', true);
				// 		} else {
				// 			$self.unifyCloseEffect(target, targetClass, bodyClass);
				// 			$.cookie('isSideMenuMini', '');
				// 		}
				// 	});
				// } else {
					$this.on('click', function(e) {
						e.preventDefault();

						if (!$(target).hasClass('toggled')) {
							$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
							$.cookie('isSideMenuMini', true);
						} else {
							$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
							$.cookie('isSideMenuMini', '');
						}
					});

					$(target).on('mouseenter', function() {
						if (!$(target).hasClass('toggled')) {
							$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
							$.cookie('isSideMenuMini', true);
						} else {
							$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
							$.cookie('isSideMenuMini', '');
						}
					});
				// }

				$(target).find('[data-hssm-target]').on('click', function(e) {
					e.preventDefault();

					var itemTarget = $(this).data('hssm-target'),
						$itemParent = $(this).parent();

					if (isShowAllExceptThis) {
						if (!$('body').hasClass('u-side-nav-mini')) {
							$itemParent.parent().find('> li:not(".has-active") > ul:not("' + itemTarget + '")').slideUp(400, function() {
								$(this).parent().removeClass('u-side-nav-opened');
								$closedItems.push(itemTarget);
							});
						} else {
							$itemParent.parent().find('> li > ul:not("' + itemTarget + '")').slideUp(400, function() {
								$(this).parent().removeClass('u-side-nav-opened');
								$closedItems.push(itemTarget);
							});
						}
					} else {
						if (!$(this).parent().hasClass('u-side-nav-opened')) {
							$closedItems.push(itemTarget);
						} else {
							$closedItems = $.grep($closedItems, function(value) {
								return value != itemTarget;
							});
						}
					}

					// $(itemTarget).slideToggle(400, function() {
					// 	$(this).parent().toggleClass('u-side-nav-opened');
					// });
					$(itemTarget).slideToggle(400)
					.parent().toggleClass('u-side-nav-opened');
				});

				// if (windW <= 992 || isSideMenuMini == true) {
				// 	$(this).trigger('click');
				// }
				//
				// $(window).on('resize', function() {
				// 	var windW = $(window).width();
				//
				// 	if (windW <= 992) {
				// 		$this.removeClass('once-opened');
				//
				// 		if (!$this.hasClass('is-active')) {
				// 			if (!$this.hasClass('once-closed')) {
				// 				$this.addClass('is-active was-opened once-closed');
				//
				// 				if (isUnifyEffect) {
				// 					$self.unifyOpenEffect(target, targetClass, bodyClass);
				// 				} else {
				// 					$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
				// 				}
				// 			}
				// 		}
				// 	} else {
				// 		$this.removeClass('once-closed');
				//
				// 		if ($this.hasClass('was-opened')) {
				// 			$this.removeClass('is-active was-opened');
				//
				// 			if (!$this.hasClass('once-opened')) {
				// 				$this.addClass('once-opened');
				//
				// 				if (isUnifyEffect) {
				// 					$self.unifyCloseEffect(target, targetClass, bodyClass);
				// 				} else {
				// 					$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
				// 				}
				// 			}
				// 		}
				// 	}
				// });

				//Actions
				collection = collection.add($this);
			});
		},

		// unifyOpenEffect: function(target, targetclass, bodyclass) {
		// 	$(target).addClass('toggled u-side-nav--mini-hover-on');
		// 	$(target + '.toggled').addClass(targetclass);
		// 	$('body').addClass(bodyclass);
		// },
    //
		// unifyCloseEffect: function(target, targetclass, bodyclass) {
		// 	$(target).children().hide();
		// 	$(target).removeClass(targetclass + ' toggled');
		// 	$('body').removeClass(bodyclass);
    //
		// 	setTimeout(function() {
		// 		$(target).children().fadeIn(100);
		// 		$(target).removeClass('u-side-nav--mini-hover-on');
		// 	}, 300);
		// },

		defaultOpenEffect: function(target, closeditems, targetclass, bodyclass, funcs) {
			$(target).addClass('toggled u-side-nav--mini-hover-on');

			if (closeditems.length > 0) {
				var items = closeditems.toString();

				$(items).slideUp(400, function() {
					$(target + '.toggled').addClass(targetclass);
					$('body').addClass(bodyclass);
					$(items).parent().removeClass('u-side-nav-opened');
					funcs();
				});
			} else {
				$(target + '.toggled').addClass(targetclass);
				$('body').addClass(bodyclass);
				funcs();
			}
		},

		defaultCloseEffect: function(target, closeditems, targetclass, bodyclass, funcs) {
			$(target).removeClass('toggled u-side-nav--mini-hover-on');
			$(target).removeClass(targetclass + ' toggled');
			$('body').removeClass(bodyclass);

			if (closeditems.length > 0) {
				setTimeout(function() {
					$(closeditems.toString()).parent().addClass('u-side-nav-opened');
					$(closeditems.toString()).slideDown(400);
					funcs();
				}, 300);
			}
		}
	}

})(jQuery);


html/admin-template/assets/js/components/hs.side-nav.js
/**
 * Side Nav wrapper.
 *
 * @author Htmlstream
 * @version 1.0
 * @requires
 *
 */

var isSideMenuMini = $.cookie('isSideMenuMini') ? true : '',
    shouldOpenSidebarOnHover = false;

;
(function($) {
	'use strict';

	$.HSCore.components.HSSideNav = {
		/**
		 *
		 *
		 * @var Object _baseConfig
		 */
		_baseConfig: {
			afterOpen: function (){},
			afterClose: function (){}
		},

		/**
		 *
		 *
		 * @var jQuery pageCollection
		 */
		pageCollection: $(),

		/**
		 * Initialization of Side Nav wrapper.
		 *
		 * @param String selector (optional)
		 * @param Object config (optional)
		 *
		 * @return jQuery pageCollection - collection of initialized items.
		 */
		init: function(selector, config) {
			this.collection = selector && $(selector).length ? $(selector) : $();
			if (!$(selector).length) return;

			this.config = config && $.isPlainObject(config) ?
				$.extend({}, this._baseConfig, config) : this._baseConfig;

			this.config.itemSelector = selector;

			this.initSideNav();

			return this.pageCollection;
		},

		initSideNav: function() {
			//Variables
			var $self = this,
				config = $self.config,
				collection = $self.pageCollection;

			//Actions
			this.collection.each(function(i, el) {
				//Variables
				var $this = $(el),
					target = $this.data('hssm-target'),
					targetClass = $this.data('hssm-class'),
					bodyClass = $this.data('hssm-body-class'),
					isUnifyEffect = Boolean($this.data('hssm-fade-effect')),
					isShowAllExceptThis = Boolean($this.data('hssm-is-close-all-except-this')),
					$closedItems = [],
					windW = $(window).width();

				$(target).find('[data-hssm-target]').each(function(i) {
					if ($(this).parent().hasClass('u-side-nav--top-level-menu-item u-side-nav-opened')) {
						$closedItems.push($(this).data('hssm-target'));
					}
				});

				// if (isUnifyEffect) {
				// 	$this.on('click', function(e) {
				// 		e.preventDefault();
        //
				// 		if (!$(target).hasClass('toggled')) {
				// 			$self.unifyOpenEffect(target, targetClass, bodyClass);
				// 			$.cookie('isSideMenuMini', true);
				// 		} else {
				// 			$self.unifyCloseEffect(target, targetClass, bodyClass);
				// 			$.cookie('isSideMenuMini', '');
				// 		}
				// 	});
				// } else {
					$this.on('click', function(e) {
						e.preventDefault();

						if (!$(target).hasClass('toggled')) {
							$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
							$.cookie('isSideMenuMini', true);

							shouldOpenSidebarOnHover = true;
						} else {
							$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
							$.cookie('isSideMenuMini', '');

							shouldOpenSidebarOnHover = false;
						}
					});
				// }

				if ($('body').hasClass('has-fixed-sidebar')) {
					$(target).on('mouseenter', function(e) {
						e.preventDefault();
						if (shouldOpenSidebarOnHover) {
							$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
						}
					});
					$(target).on('mouseleave', function(e) {
						e.preventDefault();
						if (shouldOpenSidebarOnHover) {
							$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
						}
					});
				}

				$(target).find('[data-hssm-target]').on('click', function(e) {
					e.preventDefault();

					var itemTarget = $(this).data('hssm-target'),
						$itemParent = $(this).parent();

					if (isShowAllExceptThis) {
						if (!$('body').hasClass('u-side-nav-mini')) {
							$itemParent.parent().find('> li:not(".has-active") > ul:not("' + itemTarget + '")').slideUp(400, function() {
								$(this).parent().removeClass('u-side-nav-opened');
								$closedItems.push(itemTarget);
							});
						} else {
							$itemParent.parent().find('> li > ul:not("' + itemTarget + '")').slideUp(400, function() {
								$(this).parent().removeClass('u-side-nav-opened');
								$closedItems.push(itemTarget);
							});
						}
					} else {
						if (!$(this).parent().hasClass('u-side-nav-opened')) {
							$closedItems.push(itemTarget);
						} else {
							$closedItems = $.grep($closedItems, function(value) {
								return value != itemTarget;
							});
						}
					}

					// $(itemTarget).slideToggle(400, function() {
					// 	$(this).parent().toggleClass('u-side-nav-opened');
					// });
					$(itemTarget).slideToggle(400)
					.parent().toggleClass('u-side-nav-opened');
				});

				// if (windW <= 992 || isSideMenuMini == true) {
				// 	$(this).trigger('click');
				// }
				//
				// $(window).on('resize', function() {
				// 	var windW = $(window).width();
				//
				// 	if (windW <= 992) {
				// 		$this.removeClass('once-opened');
				//
				// 		if (!$this.hasClass('is-active')) {
				// 			if (!$this.hasClass('once-closed')) {
				// 				$this.addClass('is-active was-opened once-closed');
				//
				// 				if (isUnifyEffect) {
				// 					$self.unifyOpenEffect(target, targetClass, bodyClass);
				// 				} else {
				// 					$self.defaultOpenEffect(target, $closedItems, targetClass, bodyClass, config.afterOpen);
				// 				}
				// 			}
				// 		}
				// 	} else {
				// 		$this.removeClass('once-closed');
				//
				// 		if ($this.hasClass('was-opened')) {
				// 			$this.removeClass('is-active was-opened');
				//
				// 			if (!$this.hasClass('once-opened')) {
				// 				$this.addClass('once-opened');
				//
				// 				if (isUnifyEffect) {
				// 					$self.unifyCloseEffect(target, targetClass, bodyClass);
				// 				} else {
				// 					$self.defaultCloseEffect(target, $closedItems, targetClass, bodyClass, config.afterClose);
				// 				}
				// 			}
				// 		}
				// 	}
				// });

				//Actions
				collection = collection.add($this);
			});
		},

		// unifyOpenEffect: function(target, targetclass, bodyclass) {
		// 	$(target).addClass('toggled u-side-nav--mini-hover-on');
		// 	$(target + '.toggled').addClass(targetclass);
		// 	$('body').addClass(bodyclass);
		// },
    //
		// unifyCloseEffect: function(target, targetclass, bodyclass) {
		// 	$(target).children().hide();
		// 	$(target).removeClass(targetclass + ' toggled');
		// 	$('body').removeClass(bodyclass);
    //
		// 	setTimeout(function() {
		// 		$(target).children().fadeIn(100);
		// 		$(target).removeClass('u-side-nav--mini-hover-on');
		// 	}, 300);
		// },

		defaultOpenEffect: function(target, closeditems, targetclass, bodyclass, funcs) {
			$(target).addClass('toggled u-side-nav--mini-hover-on');

			if (closeditems.length > 0) {
				var items = closeditems.toString();

				$(items).slideUp(400, function() {
					$(target + '.toggled').addClass(targetclass);
					$('body').addClass(bodyclass);
					$(items).parent().removeClass('u-side-nav-opened');
					funcs();
				});
			} else {
				$(target + '.toggled').addClass(targetclass);
				$('body').addClass(bodyclass);
				funcs();
			}
		},

		defaultCloseEffect: function(target, closeditems, targetclass, bodyclass, funcs) {
			$(target).removeClass('toggled u-side-nav--mini-hover-on');
			$(target).removeClass(targetclass + ' toggled');
			$('body').removeClass(bodyclass);

			if (closeditems.length > 0) {
				setTimeout(function() {
					$(closeditems.toString()).parent().addClass('u-side-nav-opened');
					$(closeditems.toString()).slideDown(400);
					funcs();
				}, 300);
			}
		}
	}

})(jQuery);
